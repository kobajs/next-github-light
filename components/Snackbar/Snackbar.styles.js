import styled from 'styled-components';

import MUIIconButton from '@material-ui/core/IconButton';
import MUICloseIcon from '@material-ui/icons/Close';

export const snackbarIconStyles = {
  fontSize: '20px',
  opacity: 0.9,
  marginRight: '8px',
};

export const MessageWrapper = styled.div.attrs({
  id: 'client-snackbar',
})`
  display: flex;
  align-items: center;
`;

export const IconButton = styled(MUIIconButton).attrs({
  color: 'inherit',
})``;

export const CloseIcon = styled(MUICloseIcon)`
  font-size: 20px;
  opacity: 0.9;
  margin-right: 8px;
`;
