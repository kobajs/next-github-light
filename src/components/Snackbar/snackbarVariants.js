import styled from 'styled-components';

import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import ErrorIcon from '@material-ui/icons/Error';
import InfoIcon from '@material-ui/icons/Info';
import WarningIcon from '@material-ui/icons/Warning';

import green from '@material-ui/core/colors/green';
import amber from '@material-ui/core/colors/amber';

export const snackbarVariants = {
  common: 'common',
  success: 'success',
  warning: 'warning',
  error: 'error',
  info: 'info',
};

const variantIcons = {
  common: ErrorIcon,
  success: CheckCircleIcon,
  warning: WarningIcon,
  error: ErrorIcon,
  info: InfoIcon,
};

export const snackbarVariantStyles = theme => ({
  root: {
    [theme.breakpoints.up('sm')]: {
      position: 'fixed',
      transform: 'none',
      display: 'block',
      top: 80,
      left: 'auto',
      right: 16,
      borderRadius: 8,
    },
  },
  snackbarContent: {
    flexWrap: 'nowrap',
  },
  common: {},
  success: {
    backgroundColor: green[600],
  },
  error: {
    backgroundColor: theme.palette.error.dark,
  },
  info: {
    backgroundColor: theme.palette.primary.dark,
  },
  warning: {
    backgroundColor: amber[700],
  },
});

export const getVariantIcon = variant => {
  return styled(variantIcons[variant])`
    font-size: 20px;
    opacity: 0.9;
    margin-right: 8px;
  `;
};
