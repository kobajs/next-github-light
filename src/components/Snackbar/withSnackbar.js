import React, { Fragment } from 'react';
import { connect } from 'react-redux';

import Snackbar  from '.';

import { closeSnackbar, cleanSnackbar, createSnackbarMessage } from '../../redux/actions/snackbarTypes';

export function withSnackbar(Component) {
  class WrappedComponent extends React.Component {
    // componentWillUnmount() {
    //   const { cleanSnackbar } = this.props;
    //   cleanSnackbar();
    // }

    onClose = () => {
      const { closeSnackbar } = this.props;
      closeSnackbar();
    };

    render() {
      const { children, open, message, variant, ...otherProps } = this.props;

      return (
        <Fragment>
          <Component {...otherProps}>{children}</Component>
          <Snackbar open={open} message={message} variant={variant} onClose={this.onClose} />
        </Fragment>
      );
    }
  }

  const mapStateToProps = state => ({
    open: state.snackbar.open,
    message: state.snackbar.message,
    variant: state.snackbar.variant,
  });

  const mapDispatchToProps = dispatch => ({
    closeSnackbar: () => dispatch(closeSnackbar()),
    cleanSnackbar: () => dispatch(cleanSnackbar()),
    createSnackbarMessage: (message, variant) => dispatch(createSnackbarMessage(dispatch)(message, variant)),
  });

  return connect(
    mapStateToProps,
    mapDispatchToProps,
  )(WrappedComponent);
}
