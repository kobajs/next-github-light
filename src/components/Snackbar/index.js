import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'recompose';
import classNames from 'classnames';
import values from 'lodash/values';

import { withStyles } from '@material-ui/core/styles';
import withWidth, { isWidthUp } from '@material-ui/core/withWidth';
import Typography from '@material-ui/core/Typography';
import MUISnackbar from '@material-ui/core/Snackbar';
import MUISnackbarContent from '@material-ui/core/SnackbarContent';

import { MessageWrapper, IconButton, CloseIcon, snackbarIconStyles } from './Snackbar.styles';
import { getVariantIcon, snackbarVariants, snackbarVariantStyles } from './snackbarVariants';

function Snackbar({ classes, open, message, variant, onClose, width, ...other }) {
  const Icon = getVariantIcon(variant);

  const handleClose = (event, reason) => {
    // if (reason === 'clickaway') return;

    onClose();
  };

  const direction = isWidthUp('sm', width) ? 'left' : 'up';

  return (
    <MUISnackbar
      open={open}
      onClose={handleClose}
      autoHideDuration={6000}
      className={classes.root}
      TransitionProps={{ direction }}
      anchorOrigin={{
        vertical: 'top',
        horizontal: 'right',
      }}
      disableWindowBlurListener
    >
      <MUISnackbarContent
        className={classNames(classes.snackbarContent, classes[variant])}
        aria-describedby="client-snackbar"
        message={
          <MessageWrapper>
            <Icon style={snackbarIconStyles} />
            <Typography color="inherit">{message}</Typography>
          </MessageWrapper>
        }
        action={[
          <IconButton key="close" aria-label="Close" onClick={onClose}>
            <CloseIcon />
          </IconButton>,
        ]}
        {...other}
      />
    </MUISnackbar>
  );
};

Snackbar.propTypes = {
  open: PropTypes.bool.isRequired,
  message: PropTypes.node.isRequired,
  onClose: PropTypes.func,
  variant: PropTypes.oneOf(values(snackbarVariants)),
};

Snackbar.defaultProps = {
  onClose: () => {},
  variant: snackbarVariants.common,
};

export default compose(
  withWidth(),
  withStyles(snackbarVariantStyles),
)(Snackbar);
