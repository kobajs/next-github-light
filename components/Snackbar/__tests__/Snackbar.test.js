import React from "react";
import renderer from "react-test-renderer";

import Snackbar from "..";
import { snackbarVariants } from "../snackbarVariants";

const props = {
  classes: {},
  open: false,
  message: 'SnackbarTest',
  variant: snackbarVariants.common,
  onClose: jest.fn(),
  width: 'xs',
};

test("Snackbar renders correctly", () => {
  const tree = renderer.create(<Snackbar {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
