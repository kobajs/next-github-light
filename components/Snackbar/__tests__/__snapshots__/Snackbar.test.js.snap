// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar renders correctly 1`] = `null`;
